airbyte-cdk/python/airbyte_cdk/test/entrypoint_wrapper.py

```diff
@@ -145,6 +146,30 @@ def _run_command(source: Source, args: List[str], expecting_exception: bool = Fa
     return EntrypointOutput(messages + captured_logs, uncaught_exception)
 
 
+def _get_catalog(args: List[str]) -> Mapping[str, Any]:
+    try:
+        catalog = AirbyteEntrypoint.extract_catalog(args)
+        return BaseConnector.read_config(catalog)
+    except Exception as exception:
+        print("Printing unexpected error from entrypoint_wrapper")
+        print("".join(traceback.format_exception(None, exception, exception.__traceback__)))
+
+
+def get_catalog(config: Mapping[str, Any], catalog: ConfiguredAirbyteCatalog):
```

Comment:
Why would we like to have the ConfiguredAirbyteCatalog as a Mapping? Why not use the ConfiguredAirbyteCatalog directly? Can you provide an example of usage?

Reply:
Oh here I just wanted to check that the AirbyteEntrypoint correctly parsed the configured catalog with json schema that was added using `with_json_schema` method. But yeah I agree it looks a little bit redundant, would  you prefer we remove this check?

```
    entrypoint_catalog = get_catalog(config=config ,catalog=catalog)
    assert entrypoint_catalog["streams"][0]["stream"]["json_schema"] == json_schema
```

Reply:
I would like to avoid to expose methods that return non-complex types like Mapping as it is harder to maintain. If we really want to test that, I would move this as part of the test and not expose this as part of the library.

That being said, I would assume that we don't need to test our builders as the risk of error is low and therefore the cost of maintaining such tests is too costly for the benefits. I'm willing to be convinced otherwise though so feel free to push back on this

Reply:
Ok, it makes sense to save the effort to test builder; I think I'm fine just having a test that builds the catalog using ConfiguredAirbyteStreamBuilder() + with_json_schema.

Reply:
Ok, updated, I will just check the json_schema is there after build.